### ISSUE TITLE
Wrong indexing during parsing MGF titles and retention times

### ISSUE THREAD

**Issue by caetera** [compomics/ms2rescore#80]
Hello,

`retention_times` dictionary returned by `parse_mgf_title_rt` function has scan order index as keys, thus, mapping in `peprec_from_pin` function fails with `Could not map all MGF retention times to spectrum indices.` error message.

Indexing with spectrum titles seems to solve the problem:
```
#current code
            if line[0] == "R":
                if line.startswith("RTINSECONDS="):
                    retention_times[index] = float(line[12:].strip())

#new code
            if line[0] == "R":
                if line.startswith("RTINSECONDS="):
                    retention_times[titles[index]] = float(line[12:].strip())
```

This change has been introduced at some point after v 2.0.0 (see #79). Earlier mapping in `peprec_from_pin` where performed in the opposite order and, thus, were still successful.

**Comment by ArthurDeclercq** [comment]
Hi @caetera,

Thank you for the comment, indeed there was a bug introduced in the newer versions causing this to break. Thank you for suggesting the solution by using earlier version of MS²Rescore to still be able to run this. We will fix this in the next version of MS²Rescore. 

Thank you!
Cheers,
Arthur

**Comment by RalfG** [comment]
Hi @caetera, Happy to say that this should now fixed in v3.0 of MS²Rescore.
